File: Php/app/User/Details/UserDetailsModel.php
<?php
namespace app\User\Details{

	use Domain\Classes\User;
	use core\PageModel;
	use app\Urls\ChangePasswordUrlModel;
	use app\Urls\UserEditUrlModel;
	use app\User\Avatar\AvatarModelBuilder;
	use integration\Avatar\AvatarService;

	class UserDetailsModel extends PageModel {

		public $userName;
		public $displayName;
		public $realName;
		public $email;
		public $editLink;
		public $changePasswordLink;
		public $showEditLink;
		public $showPasswordLink;
		public $avatarModel;

		public function __construct(User $currentUser, User $displayUser, AvatarService $avatarService){
			parent::__construct($currentUser);
			$this->userName = $displayUser->getUserName();
			$this->displayName = $displayUser->getDisplayName();
			$this->realName = $displayUser->getRealName();
			$this->email = $displayUser->getEmail();

			$avatarModelBuilder = new AvatarModelBuilder($avatarService);
			$this->avatarModel = $avatarModelBuilder->build($displayUser->getEmail());

			$this->showEditLink = false;
			$this->showPasswordLink = false;
			$isViewingCurrentUser = $displayUser->getUserName() == $currentUser->getUserName();

			if($currentUser->isAdmin() || $isViewingCurrentUser){
				$this->showEditLink = true;
				$this->editLink = new UserEditUrlModel($displayUser);
			}

			if($isViewingCurrentUser){
				$this->showPasswordLink = true;
				$this->changePasswordLink = new ChangePasswordUrlModel();
			}
		}

	}

}

File: Web/Models/UserModels/UserDetailsModel.cs
using Core.Classes;
using Core.Services;
using Web.ModelFactories.MiscModelFactories;
using Web.Models.MiscModels;
using Web.Models.PageBaseModels;
using Web.Models.UrlModels;

namespace Web.Models.UserModels{

	public class UserDetailsModel : PageProperties {

		public string UserName { get; set; }
		public string DisplayName { get; set; }
		public string RealName { get; set; }
		public string Email { get; set; }
		public UrlModel EditLink { get; set; }
		public UrlModel ChangePasswordLink { get; set; }
		public bool ShowEditLink { get; set; }
		public bool ShowPasswordLink { get; set; }
		public AvatarModel AvatarModel { get; set; }

		public UserDetailsModel(User currentUser, User displayUser, IAvatarService avatarService)
            : base(currentUser)
        {
			UserName = displayUser.UserName;
			DisplayName = displayUser.DisplayName;
			RealName = displayUser.RealName;
			Email = displayUser.Email;

			var avatarModelBuilder = new AvatarModelFactory(avatarService);
			AvatarModel = avatarModelBuilder.Create(displayUser.Email);

			ShowEditLink = false;
			ShowPasswordLink = false;
			var isViewingCurrentUser = displayUser.UserName == currentUser.UserName;

			if(currentUser.IsAdmin || isViewingCurrentUser){
				ShowEditLink = true;
				EditLink = new UserEditUrlModel(displayUser);
			}

			if(isViewingCurrentUser){
				ShowPasswordLink = true;
				ChangePasswordLink = new ChangePasswordUrlModel();
			}
		}

        public override string BrowserTitle
        {
            get { return "User Details"; }
        }

	}

}